### PR TITLE
Fix typescript type errors

### PR DESCRIPTION
**Pull request by mgunnerud**
## Description

Fix type issues reported by Typescript. Other in-depth explanations are added as comments. Two errors reported by typescript is not fixed by this PR, but should be fixed in https://github.com/Altinn/altinn-access-management-frontend/issues/985 instead.

## Related Issue(s)
- #972

## Verification
- [ ] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/components/CollectionBar/CollectionBar.tsx [hunk 1]
```diff
@@ -73,7 +73,7 @@ export const CollectionBar = ({
             <Button
               variant='tertiary'
               size='small'
-              color={color === 'dark' ? 'inverted' : undefined}
+              className={cn(classes.buttonText, classes[color])}
```
Comment (mgunnerud) [inline src/components/CollectionBar/CollectionBar.tsx:76]: `inverted` color for \<Button\> is deprecated. Use new css class instead to set text color.

File: src/features/apiDelegation/offered/ChooseApiPage/ApiSearchResult.tsx [hunk 2]
```diff
@@ -13,9 +13,10 @@ import { ErrorPanel } from '@/components';
 import { ApiActionBar } from '../../components/ApiActionBar';
 
 import classes from './ChooseApiPage.module.css';
+import type { CustomError } from '@/dataObjects';
 
 interface ApiSearchResultsProps {
-  error?: FetchBaseQueryError | SerializedError;
+  error?: FetchBaseQueryError | SerializedError | CustomError;
```
Comment (mgunnerud) [inline src/features/apiDelegation/offered/ChooseApiPage/ApiSearchResult.tsx:19]: Neither of the types `FetchBaseQueryError` or `SerializedError` has the `statusCode` property. Added a third type which has both `message` and `statusCode` properties.

File: src/rtk/features/apiDelegation/overviewOrg/overviewOrgSlice.ts [hunk 3]
```diff
@@ -334,12 +343,13 @@ const overviewOrgSlice = createSlice({
         state.loading = false;
       })
       .addCase(fetchOverviewOrgsReceived.rejected, (state, action) => {
+        const payload = action.payload as RejectedPayload;
         state.loading = true;
-        state.error.statusCode = String(action.payload?.response?.status) ?? 'Unknown code';
```
Comment (mgunnerud) [inline src/rtk/features/apiDelegation/overviewOrg/overviewOrgSlice.ts:338]: The `??` makes no sense here; `String(action.payload?.response?.status)` will be either status as a string, 'undefined' or ''

**Review by allinox — APPROVED** [review]
Thank you so much for cleaning up our code! 😄✨

**Pull request merged**